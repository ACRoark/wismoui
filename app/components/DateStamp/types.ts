export type DateStampFormat = 'auto' | 'long' | 'short';
