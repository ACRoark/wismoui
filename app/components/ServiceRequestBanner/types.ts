export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
