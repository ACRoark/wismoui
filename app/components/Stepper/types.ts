import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
