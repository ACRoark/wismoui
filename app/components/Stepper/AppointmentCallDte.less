@import (reference) '../../styles/themes/index.less';

@appointment-call-dte-cls-prefix: ~'@{dte-prefix}-appointment-call-dte';

.@{appointment-call-dte-cls-prefix} {
  .message {
    color: @dte-red;
    font-weight: @dte-wismo-font-weight-semi-bold;
  }
}
