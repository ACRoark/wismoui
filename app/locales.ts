export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
