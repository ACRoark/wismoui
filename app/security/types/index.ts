import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
