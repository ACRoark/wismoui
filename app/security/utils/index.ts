import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
