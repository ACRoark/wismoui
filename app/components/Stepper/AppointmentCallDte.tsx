import React from 'react';
import { FormattedMessage } from 'react-intl';

import DteContactNumber from 'components/DteContactNumber';
import { ProductType } from 'types';

import getAppointmentType from './getAppointmentType';
import messages from './messages';

import './AppointmentCallDte.less';

interface IAppointmentCallDteProps {
  product: ProductType;
}

const AppointmentCallDte: React.FC<IAppointmentCallDteProps> = (
  props: IAppointmentCallDteProps,
): React.ReactElement => {
  const { product } = props;

  const serviceType = getAppointmentType(product);

  return (
    <div className="dte-wismo-appointment-call-dte">
      <div>
        <FormattedMessage {...messages[serviceType]} values={{ window: null }} />
      </div>
      <span className="message">
        <FormattedMessage
          {...messages.CALL_DTE}
          values={{
            phoneNumber: <DteContactNumber />,
          }}
        />
      </span>
    </div>
  );
};

export default AppointmentCallDte;
