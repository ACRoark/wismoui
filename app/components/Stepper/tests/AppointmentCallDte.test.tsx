import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { createMockStore, createSnapshotWithIntl } from 'testing/utils';
import { ProductType } from 'types';

import AppointmentCallDte from '../AppointmentCallDte';

const expectComponentToMatchSnapshot = (product: ProductType): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  languages.forEach((language: string): void => {
    const tree = createSnapshotWithIntl(<Provider store={createMockStore()}><AppointmentCallDte product={product} /></Provider>, language);

    expect(tree).toMatchSnapshot();
  });
};

describe('<AppointmentCallDte />', (): void => {

  it('should render correctly when a product is given', (): void =>
    expectComponentToMatchSnapshot('EFR_D1'));
});
