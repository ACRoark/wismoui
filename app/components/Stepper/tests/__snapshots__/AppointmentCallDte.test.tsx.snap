// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentCallDte /> should render correctly when a product is given 1`] = `
<div
  className="dte-wismo-appointment-call-dte"
>
  <div>
    Electric appointment 
  </div>
  <span
    className="message"
  >
    Please call us at 
    <a
      className="dte-wismo-dte-contact-number"
      href="tel:+8004774747"
    >
      800.477.4747
    </a>
  </span>
</div>
`;

exports[`<AppointmentCallDte /> should render correctly when a product is given 2`] = `
<div
  className="dte-wismo-appointment-call-dte"
>
  <div>
    Cita electrica 
  </div>
  <span
    className="message"
  >
    Por favor ll√°menos al 
    <a
      className="dte-wismo-dte-contact-number"
      href="tel:+8004774747"
    >
      800.477.4747
    </a>
  </span>
</div>
`;
